File: src/lib/api/import.ts
import { command } from './client';import type { ImportPreview } from './types';
export const previewImport=(json:string)=>command<ImportPreview>('preview_import',{json});
export const importVocabulary=(blockId:string,json:string)=>command<{importedCount:number}>('import_vocabulary',{blockId,json});

File: src-tauri/src/commands/import.rs
use crate::{
    AppState,
    error::AppResult,
    models::{ImportPreview, ImportResult},
    services::import_validation,
};
use tauri::State;
#[tauri::command]
pub fn preview_import(json: String) -> AppResult<ImportPreview> {
    import_validation::preview(&json)
}
#[tauri::command]
pub async fn import_vocabulary(
    state: State<'_, AppState>,
    block_id: String,
    json: String,
) -> AppResult<ImportResult> {
    import_validation::import(&state.pool, &block_id, &json).await
}
